### DISCUSSION TITLE
Switching schemas between requests

### DISCUSSION THREAD

**Discussion by priyadarshiniguptan** [PostgREST/postgrest#2745] [Q&A]
Hi,

I am looking for a solution to be able to switch between multiple schemas based on the request URI for GET and POST requests 

```
REACT_APP_GET_PROFILE_KEY='Accept-Profile'
REACT_APP_POST_PROFILE_KEY='Content-Profile'
REACT_APP_GET_PROFILE_VALUE='stateless_service_bzo'
REACT_APP_POST_PROFILE_VALUE='stateful_service_bzo'

const {
  REACT_APP_POST_PROFILE_KEY,
  REACT_APP_POST_PROFILE_VALUE,
  REACT_APP_GET_PROFILE_KEY,
  REACT_APP_GET_PROFILE_VALUE,
}:any = process.env;
const fetchJson = (url: string, options: fetchUtils.Options = {}) => {
  const customHeaders = (options.headers ||
    new Headers({
      Accept: "application/json",
    })) as Headers;
 
  customHeaders.set(
    options.method === "POST" ? REACT_APP_POST_PROFILE_KEY : REACT_APP_GET_PROFILE_KEY,
    options.method === "POST" && url.includes("insert_or_update_grouping_master")
      ? REACT_APP_GET_PROFILE_VALUE
      : options.method === "POST"
      ? REACT_APP_POST_PROFILE_VALUE
      : REACT_APP_GET_PROFILE_VALUE
  );
  options.headers = customHeaders;
```

Also I want to be able to specify multiple schemas (comma separated in the .env variables so that when I make a request from the application the corresponding postgres function gets called by automatically look for the search_path set.

Can anyone help me with this?

**Comment by wolfgangwalther** [comment]
Have you had a look at https://postgrest.org/en/stable/api.html#multiple-schemas?

Edit: Seems like you did, because your code snippet has `Accept-Profile` and `Content-Profile` in it. The code snippet is not really helpful, though, because you're not explaining what it does.
